test-suite/shared/src/test/scala/org/scalajs/testsuite/javalib/lang/DoubleTest.scala

```diff
@@ -64,31 +64,270 @@ class DoubleTest {
+    test("1e-7", 0.0000001d) // x = 1.0e-7
+
+    // minAndMax
+    test("1.7976931348623157e+308", JDouble.longBitsToDouble(0x7fefffffffffffffL))
+    // TODO: (align the output between Ryu and JS)
+    assertEquals(
+      if (executingInPureWebAssembly) "4.9e-324" else "5e-324",
+      JDouble.longBitsToDouble(1).toString
+    )
+
+    // roundingModeEven
+    test("-21098088986959630", -2.109808898695963E16)
+
+    // regressionTests
+    test("4.940656e-318", 4.940656E-318d)
+    test("1.18575755e-316", 1.18575755E-316d)
+    test("2.989102097996e-312", 2.989102097996E-312d)
+    test("9060801153433600", 9.0608011534336E15d)
+    test("4708356024711512000", 4.708356024711512E18)
+    test("9409340012568248000", 9.409340012568248E18)
+    test("1.8531501765868567e+21", 1.8531501765868567E21)
+    test("-3.347727380279489e+33", -3.347727380279489E33)
+    test("19430376160308388", 1.9430376160308388E16)
+    test("-69741824662760956000", -6.9741824662760956E19)
+    test("4381605060114783700", 4.3816050601147837E18)
+
+    // LotsOfTrailingZeros
+    test("2.9802322387695312e-8", 2.98023223876953125E-8)
+
+    // LooksLikePow5
+    // These numbers have a mantissa that is a multiple of the largest power of 5 that fits,
+    // and an exponent that causes the computation for q to result in 22, which is a corner
+    // case for Ryu.
+    test("5.764607523034235e+39", JDouble.longBitsToDouble(0x4830F0CF064DD592L))
+    test("1.152921504606847e+40", JDouble.longBitsToDouble(0x4840F0CF064DD592L))
+    test("2.305843009213694e+40", JDouble.longBitsToDouble(0x4850F0CF064DD592L))
+
+    // OutputLength
+    test("1.2", 1.2)
+    test("1.23", 1.23)
+    test("1.234", 1.234)
+    test("1.2345", 1.2345)
+    test("1.23456", 1.23456)
+    test("1.234567", 1.234567)
+    test("1.2345678", 1.2345678) // already tested in Regression
+    test("1.23456789", 1.23456789)
+    test("1.234567895", 1.234567895) // 1.234567890 would be trimmed
+    test("1.2345678901", 1.2345678901)
+    test("1.23456789012", 1.23456789012)
+    test("1.234567890123", 1.234567890123)
+    test("1.2345678901234", 1.2345678901234)
+    test("1.23456789012345", 1.23456789012345)
+    test("1.234567890123456", 1.234567890123456)
+    test("1.2345678901234567", 1.2345678901234567)
+    // Test 32-bit chunking
+    test("4.294967294", 4.294967294) // 2^32 - 2
+    test("4.294967295", 4.294967295) // 2^32 - 1
+    test("4.294967296", 4.294967296) // 2^32
+    test("4.294967297", 4.294967297) // 2^32 + 1
+    test("4.294967298", 4.294967298) // 2^32 + 2
+
+    // MinMaxShift
+    // Test min, max shift values in shiftright128
+    val maxMantissa = (1L << 53) - 1
+    // 32-bit opt-size=0:  49 <= dist <= 50
+    // 32-bit opt-size=1:  30 <= dist <= 50
+    // 64-bit opt-size=0:  50 <= dist <= 50
+    // 64-bit opt-size=1:  30 <= dist <= 50
+    assertEquals("1.7800590868057611e-307", ieeeParts2Double(sign = false, 4, 0).toString)
```

Comment:
```suggestion
    test("1.7800590868057611e-307", ieeeParts2Double(sign = false, 4, 0))
```
and everywhere below?